segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -306,7 +311,14 @@ protected void doStart() {
         }
 
         // Delayed start. Secondary services need not be started in order for us to accept requests.
-        delayedStart.thenComposeAsync(v -> startSecondaryServicesAsync(), this.executor)
+        delayedStart
+                .thenComposeAsync( v -> {
+                    if (this.storage instanceof ChunkedSegmentStorage) {
```

Comment:
Please roll this into `startSecondaryServicesAsync`

Reply:
ok makes sense

Reply:
on second thought, finishing storage bootstrap is a separate and pre-requisite step for any other secondary service to work.